internal/api/kubernetes/client.go

```diff
@@ -88,87 +92,35 @@ func (c client) GetWorkloads(namespaces []string, resourceTypes []string, ctx co
 	return nil
 }
 
 // UpscaleWorkload upscales the workload to the original replicas
 func (c client) UpscaleWorkload(workload scalable.Workload, ctx context.Context) error {
-	currentReplicas, err := workload.GetCurrentReplicas()
-	if err != nil {
-		return fmt.Errorf("failed to get current replicas for workload: %w", err)
-	}
-	originalReplicas, err := c.getOriginalReplicas(workload)
+	err := workload.ScaleUp()
 	if err != nil {
-		return fmt.Errorf("failed to get original replicas for workload: %w", err)
-	}
-	if originalReplicas == values.Undefined {
-		slog.Debug("original replicas is not set, skipping", "workload", workload.GetName(), "namespace", workload.GetNamespace())
-		return nil
-	}
-	if originalReplicas == currentReplicas {
-		slog.Debug("workload is already at original replicas, skipping", "workload", workload.GetName(), "namespace", workload.GetNamespace())
-		return nil
+		return fmt.Errorf("failed to complete the scale up process on workload: %w", err)
 	}
-
-	workload.SetReplicas(originalReplicas)
-	c.removeOriginalReplicas(workload)
-	err = workload.Update(c.clientset, ctx)
+	err = workload.Update(c.clientset, c.dynamicClient, ctx)
 	if err != nil {
-		return fmt.Errorf("failed to update workload: %w", err)
+		return fmt.Errorf("failed to update the workload after completing the scaling up process: %w", err)
 	}
 	slog.Debug("successfully scaled up workload", "workload", workload.GetName(), "namespace", workload.GetNamespace())
 	return nil
 }
 
-// setOriginalReplicas sets the original replicas annotation on the workload
-func (c client) setOriginalReplicas(originalReplicas int, workload scalable.Workload) {
-	annotations := workload.GetAnnotations()
-	if annotations == nil {
-		annotations = map[string]string{}
-	}
-	annotations[annotationOriginalReplicas] = fmt.Sprintf("%d", originalReplicas)
-	workload.SetAnnotations(annotations)
-}
-
-// getOriginalReplicas gets the original replicas annotation on the workload. nil is undefined
-func (c client) getOriginalReplicas(workload scalable.Workload) (int, error) {
-	annotations := workload.GetAnnotations()
-	originalReplicasString, ok := annotations[annotationOriginalReplicas]
-	if !ok {
-		return values.Undefined, nil
-	}
-	originalReplicas, err := strconv.Atoi(originalReplicasString)
-	if err != nil {
-		return 0, fmt.Errorf("failed to parse original replicas annotation on workload: %w", err)
-	}
-	return originalReplicas, nil
-}
-
-func (c client) removeOriginalReplicas(workload scalable.Workload) {
-	annotations := workload.GetAnnotations()
-	delete(annotations, annotationOriginalReplicas)
-	workload.SetAnnotations(annotations)
-}
-
 // AddErrorEvent creates or updates a new event on the workload
 func (c client) AddErrorEvent(reason, id, message string, workload scalable.Workload, ctx context.Context) error {
-	hash := sha1.Sum([]byte(fmt.Sprintf("%s.%s", id, message)))
+	hash := sha256.Sum256([]byte(fmt.Sprintf("%s.%s", id, message)))
```

Comment:
oh yea i forgot about that, it was fixed in other PRs but they aren't merged yet because the other code owners were on vacation

Reply:
That was the reason why I disabled the lint. I didn't want to modify this class before it was merged on main